aduro/model.py

```diff
@@ -0,0 +1,239 @@
+"""Data classes for Aduro"""
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, List
+
+from .helpers import try_convert_object
+
+TIMESTAMP_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+# pylint: disable=unsupported-binary-operation
+
+
+@dataclass
+class Connection:
+    """Dataclass for device connection"""
+
+    timestamp: datetime
+    online: bool
+
+    @staticmethod
+    def from_dict(obj: Any) -> "Connection":
```

Comment:
Maybe I'm not 100% sure on the syntax, but shouldn't it be like this?

```suggestion
    def from_dict(obj: Any) -> Connection:
```

Reply:
@Tenrec :: Now I why I should have left comment on this when I had to change it, because now I don't remember anymore why it had to be in quotes 😆 
I'm changing this to `pydantic.dataclasses` to let this be handled by module.

Reply:
Then I'll hold back on any more comments on the dataclasses until after this rewrite. :)

Reply:
@Tenrec now the dataclasses are converted to `pydantic`. It does look better, but I have not found yet how to convert the `data` field in `State` to common, since it always comes as string from api. Have to dig into theyr code to find out how to use custom converter. In worse case I use custom type and convert that on init with my converter 😄